### PR TITLE
Fix to #17066 - Query: Translate ElementAt(OrDefault)

### LINKED ISSUE #17066: Query: Translate ElementAt(OrDefault)

### PR DESCRIPTION
**Pull request by maumar**
Enable member pushdown for ElementAt(OrDefault) and then handle the method in translation in a similar way to First(OrDefault) with an optional Skip(x) element.

Fixes #17066

### CODE REVIEW

File: src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs [hunk 1]
```diff
@@ -346,13 +346,29 @@ protected override Expression VisitBinary(BinaryExpression binaryExpression)
...
+                    || genericMethod == QueryableMethods.LastWithPredicate
+                    || genericMethod == QueryableMethods.LastOrDefaultWithPredicate)
                 {
                     source = Expression.Call(
                         QueryableMethods.Where.MakeGenericMethod(source.Type.GetSequenceType()),
                         source,
                         nonNullMethodCallExpression.Arguments[1]);
                 }
+                else if ((genericMethod == QueryableMethods.ElementAt || genericMethod == QueryableMethods.ElementAtOrDefault)
+                    && (nonNullMethodCallExpression.Arguments[1] is not ConstantExpression constantIndex
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs:363]: Why this check is needed?

Comment (maumar) [reply src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs:363]: just a small optimization, so we don't add skip (0) needlessly

Comment (smitpatel) [reply src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs:363]: then you should do it in TranslateSkip.

Comment (maumar) [reply src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs:363]: removed the optimization altogether, there are some cases when we want to keep Skip(0) to force pushdown, so we would want to keep that probably.

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**